### PR TITLE
fix: donation banner tweaks

### PR DESCRIPTION
**Pull request by sultanowski**
### What

1. Replace checkbox and text to [X] button.
2. Better display on both mobile and desktop.
3. Refactor of `document.cookie` code and placed it directly in `donate_banner.tt.html` file.
4. Set the banner cookie time to 14 days.

### Screenshot

https://imgur.com/a/3KPz3Ov

### Related issue(s) and discussion

- #7412 
- #5720

### CODE REVIEW

File: templates/web/common/includes/donate_banner.tt.html [hunk 1]
```diff
@@ -3,54 +3,69 @@
...
+    </div>
+    <div class="banner__close">
+      <span id="hide-donate-banner" onclick="DonationButton();" class="material-icons banner__close--icon">close</span>
+    </div>
+	</div>
+</section>
+
+<script>
+  let d = new Date();
+  let expires = 'expires=' + d.toUTCString();
```
Comment (alexgarel) [inline templates/web/common/includes/donate_banner.tt.html:26]: it seems we don't need this any-more ?

File: templates/web/common/includes/donate_banner.tt.html [hunk 2]
```diff
@@ -3,54 +3,69 @@
...
+  let bannerID = document.getElementById('donate-banner');
+  let bcval = "1";
```
Comment (alexgarel) [inline templates/web/common/includes/donate_banner.tt.html:28]: it seems we don't need this any-more ?

File: templates/web/common/includes/donate_banner.tt.html [hunk 3]
```diff
@@ -3,54 +3,69 @@
...
+      let c = ca[i];
+      while (c.charAt(0) == ' ') { c = c.substring(1); }
+      if (c.indexOf(name) == 0) { return c.substring(name.length, c.length); }
+    }
+    return '';
+  }
+
+  function DonationButton() {
+    deleteBannerCookie('off-donation-banner');
+    setBannerCookie('off-donation-banner', 1, 14);
```
Comment (alexgarel) [inline templates/web/common/includes/donate_banner.tt.html:58]: I seems to me 14 days is not enough, it's a bit pushy to users.

If I donate, I would expect this not to show up again for at least 180 days !

If possible, I would set it on whole openfoodfacts.xx subdomain (not only for eg. world) (that is you can set domain to the last two components of domain).

@teolemon what is your though ?

**Review by alexgarel — COMMENTED** [review]
@sultanowski really cool !

I tested it, it works for me.

I think there is code in [Display.pm (line 7621)](https://github.com/sultanowski/openfoodfacts-server/blob/sultanowski-donation-banner/lib/ProductOpener/Display.pm#L7621) that should be cleaned up.

I wait for some suggestion resolutions to approve.

File: templates/web/common/includes/donate_banner.tt.html [hunk 4]
```diff
@@ -3,54 +3,69 @@
...
+  let d = new Date();
+  let expires = 'expires=' + d.toUTCString();
+  let bannerID = document.getElementById('donate-banner');
+  let bcval = "1";
+
+  function setBannerCookie(bcname, bcval, bcexdays) {
+    d.setTime(d.getTime() + (bcexdays*60*60*24*1000));
+    let expires = 'expires=' + d.toUTCString();
+    document.cookie = bcname + '=' + bcval + ';' + expires + ';path=/';
+    console.log(expires);
```
[the remark this answers is not in the document]
Comment (sultanowski) [reply templates/web/common/includes/donate_banner.tt.html:34]: Oops, forgot about those! Thank you, I will remove it soon, also unused vars and logs.

File: templates/web/common/includes/donate_banner.tt.html [hunk 5]
```diff
@@ -3,54 +3,69 @@
...
+  }
+
+
+  function deleteBannerCookie(bcname) {
+    d.setTime(d.getTime() + (60*60*24*1000));
+    let expires = 'expires=' + d.toUTCString();
+    document.cookie = bcname + '=;' + expires + ';path=/';
+  }
```
[the remark this answers is not in the document]
Comment (sultanowski) [reply templates/web/common/includes/donate_banner.tt.html:42]: It's like a double-check for me - cleans the cookie value and set expire for current time, if you feel like it is not really important I can test the code without it.

**Comment by stephanegigandet** [comment]
On mobile, the banner is now more a full screen banner than a banner:

after:
image

before:
image

And if we remove the message to close the box, and put the cross instead we would get a smaller banner:

image

**Comment by stephanegigandet** [comment]
Note that yesterday we added a 2nd donation box (identical except that it cannot be closed) at the bottom of the page, so unfortunately it's causing some conflicts.

**Comment by sultanowski** [comment]
Pushed cosmetic changes mentioned in this discussion, banner is slightly smaller than the actual one. What should I do to resolve this conflict? I didn't pulled to this branch during development and I don't want to merge it with `main` - that might be even worse I think.

**Comment by alexgarel** [comment]
@sultanowski the best would be to sync main branch on your project, update main locally, switch to your branch, then use "rebase main" (read some tutorial, it's a good command to master). Then you have to "git push --force".

**Comment by sultanowski** [comment]
@alexgarel I found solution to apply a cookie for all subdomains and all possible versions (.org, .net, etc.) now I am going to rebase and update this branch.

File: templates/web/common/includes/donate_banner.tt.html [hunk 6]
```diff
@@ -24,11 +24,14 @@
...
   let d = new Date();
   let bannerID = document.getElementById('donate-banner');
+  let getDomain = window.location.origin.split('.');
 
   function setBannerCookie(bcname, bcval, bcexdays) {
     d.setTime(d.getTime() + (bcexdays*60*60*24*1000));
     let expires = 'expires=' + d.toUTCString();
-    document.cookie = bcname + '=' + bcval + ';' + expires + ';path=/';
+    // Apply cookie for every type of link contains '.openfoodfacts.'
+    let domain = 'domain=.openfoodfacts.' + getDomain[2];
```
Comment (alexgarel) [inline templates/web/common/includes/donate_banner.tt.html:33]: Product opener is not only used by openfoodfacts but also openproductsfacts, etc. 
We should never hardcode domain.

```suggestion
    // Apply cookie for whole primary domain (xxx.org)
    let domain = 'domain=.' + getDomain.slice(-2).join(".");
```

**Review by alexgarel — CHANGES_REQUESTED** [review]
Thank you !
@sultanowski still a small problem and we are done :-)

File: templates/web/common/includes/donate_banner.tt.html (hunk 6 above)
Comment (sultanowski) [reply templates/web/common/includes/donate_banner.tt.html]: Added this change in `204edea` commit.

Comment (sultanowski) [reply templates/web/common/includes/donate_banner.tt.html]: Didn't worked, I will figure it out soon.

**Comment by sultanowski** [comment]
What I've done in small summary:
* Delete `donate_banner_bottom.tt.html` file and create new instance of it inside `donate_banner.tt.html` file.
* Replace long checkbox with label to graphical [x] button.
* Banner's cookie should work with every type of open...facts page. Now URL is dynamically taken from client's URL and it apply on close button click.
* Fix `max-width` of donation-banner content, so now it works like previously on wide screens.
* Reduce margins and paddings to make donation banner compact in size.
* Clean-up of `Display.pm` file and delete unused cookie jQuery function.

File: templates/web/common/includes/donate_banner.tt.html (hunk 6 above)
Comment (sultanowski) [reply templates/web/common/includes/donate_banner.tt.html]: I used `.slice(1)` to delete `http://world`. `.slice(-2)` caused that close button didn't worked.

**Review by alexgarel — APPROVED** [review]

**Comment by alexgarel** [comment]
Fine @sultanowski, thank you so much for your dedication !

I think we could improve styling because the banner is a bit deported on the right because of the "hand-heart" icon on the right… but that would be for another PR :-)

**Pull request merged**